src/geom/prep/PreparedPolygonPredicate.cpp

```diff
@@ -91,6 +91,11 @@ bool
 PreparedPolygonPredicate::isAnyTestComponentInTargetInterior(
     const geom::Geometry* testGeom) const
 {
+    if (testGeom->getGeometryTypeId() == GEOS_POINT) {
```

Comment:
There's a few of these in `PreparedPolygonPredicate`, right?

Another alternative would be to create a visitor utility class based on `GeometryComponentFilter`, and avoid creating the list of coordinates entirely.

Reply:
But maybe it's good enough to special-case Point, which is probably the high-value optimization.

Reply:
I also realize now that this is doing too much work when testing Polygons with holes, since it tests each ring, but only needs to test the polygon shell.

Reply:
I like your filter idea and implemented it in 3cf6bb4.

Reply:
BTW, I'm curious what you think about changing the filter return type from `void` to `bool`, where `true` means "continue iterating" or something.

Reply:
> BTW, I'm curious what you think about changing the filter return type from `void` to `bool`, where `true` means "continue iterating" or something.

In general, in favour.  But it's a breaking API change, at least in JTS.  Probably not too much external use of this, though, so perhaps worth doing.

Note that `CoordinateSequenceFilter` has an isDone method that supports short-circuiting.  Not sure why I did it that way rather than a boolean return value (perhaps the idea was to make the done state more persistent, to allow easier checking by mulitple clients).  Is it worth continuing this pattern?

Reply:
> Is it worth continuing this pattern?

Yes, calling `isDone()` seems more clear than assigning significance to the return value. I could add this to `CoordinateFilter`.

Reply:
Sure - nice that we can change GEOS API at will!  

Or... maybe wait until a use case emerges?  For this PR wouldn't it be better to add isDone to `GeometryComponentFilter`?  And then would need to change PreparedPolygonPredicate to use a filter rather than `ComponentCoordinateExtracter`.

Reply:
Well, the use case is here, where I had to essentially implement `isDone` in both of the `CoordinateFilters` I implemented in `PreparedPolygonPredicate`.

Reply:
See my latest comment on this code.  I think using `CoordinateFilter` here is not efficient?